src/main/java/ua/price/bdd_solution/bdd_pages/pages/SearchPageResults.java

```diff
@@ -0,0 +1,113 @@
+package ua.price.bdd_solution.bdd_pages.pages;
+
+import net.serenitybdd.core.annotations.findby.FindBy;
+import net.serenitybdd.core.pages.WebElementFacade;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import ua.price.bdd_solution.bdd_pages.base_page.BasePage;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SearchPageResults extends BasePage {
+    @FindBy(xpath = ".//div[contains(@class, 'product-block type')]")
+    private List<WebElementFacade> searchResultsList;
+
+    @FindBy(xpath = ".//a[contains(@class,'btn-orange')]")
+    private WebElementFacade itemPageButton;
+
+    @FindBy(xpath = ".//a[@data-producer-alias='samsung']")
+    private WebElementFacade selectProducerSamsung;
+
+    @FindBy(xpath = ".//div[@id='paginator']")
+    private WebElementFacade pagination;
+
+    @FindBy(xpath = ".//li[contains(@class,'selected')]/span")
+    private WebElementFacade selectedPage;
+
+    @FindBy(xpath = ".//li[@class='next']")
+    private WebElementFacade nextPage;
+
+    @FindBy(xpath = ".//a[@data-producer-alias='samsung']/following-sibling::span")
+    private WebElementFacade amountOfItemsForSelectedProducerSamsung;
+
+    @FindBy(xpath = ".//input[@id='price_min_']")
+    private WebElementFacade minPriceInput;
+
+    @FindBy(xpath = ".//input[@id='price_max_']")
+    private WebElementFacade maxPriceInput;
+
+    @FindBy(xpath = ".//a[contains(@class,'btn-filters-submit')]")
+    private WebElementFacade submitFiltrationButton;
+
+    @FindBy(xpath = "(.//span[contains(@class,'add-to-wishlist-link')])[4]")
```

Comment:
Can we find more stable XPath here? Without using indexes?

Reply:
Done